github/repos.go

```diff
@@ -501,29 +501,56 @@ func (s *RepositoriesService) ListTags(owner string, repo string, opt *ListOptio
 
 // Branch represents a repository branch
 type Branch struct {
-	Name       *string     `json:"name,omitempty"`
-	Commit     *Commit     `json:"commit,omitempty"`
-	Protection *Protection `json:"protection,omitempty"`
+	Name      *string `json:"name,omitempty"`
+	Commit    *Commit `json:"commit,omitempty"`
+	Protected *bool   `json:"protected,omitempty"`
 }
 
 // Protection represents a repository branch's protection
 type Protection struct {
-	Enabled              *bool                 `json:"enabled,omitempty"`
-	RequiredStatusChecks *RequiredStatusChecks `json:"required_status_checks,omitempty"`
+	RequiredStatusChecks *RequiredStatusChecks `json:"required_status_checks"`
+	Restrictions         *BranchRestrictions   `json:"restrictions"`
+}
+
+// ProtectionRequest represents a request to create/edit a branch's protection.
+// It is separate from Protection above because the request structure is
+// different from the response structure.
+type ProtectionRequest struct {
+	RequiredStatusChecks *RequiredStatusChecks      `json:"required_status_checks"`
+	Restrictions         *BranchRestrictionsRequest `json:"restrictions"`
 }
 
 // RequiredStatusChecks represents the protection status of a individual branch
 type RequiredStatusChecks struct {
-	// Who required status checks apply to.
-	// Possible values are:
-	//     off
-	//     non_admins
-	//     everyone
-	EnforcementLevel *string `json:"enforcement_level,omitempty"`
-	// The list of status checks which are required
+	// Enforce required status checks for repository administrators.
+	IncludeAdmins *bool `json:"include_admins,omitempty"`
+	// Require branches to be up to date before merging.
+	Strict *bool `json:"strict,omitempty"`
+	// The list of status checks to require in order to merge into this
+	// branch
 	Contexts *[]string `json:"contexts,omitempty"`
 }
 
+// BranchRestrictions represents the restriction that only certain users or
+// teams may push to a branch
+type BranchRestrictions struct {
+	// The list of user logins with push access
+	Users []*User `json:"users,omitempty"`
+	// The list of team slugs with push access
+	Teams []*Team `json:"teams,omitempty"`
```

Comment:
This seems inconsistent with `BranchRestrictionsRequest` below. Both structs have fields that are slices, and both use `omitempty` JSON option.

However, this struct uses fields that are slices of pointers (to `User` and `Team` structs), but `BranchRestrictionsRequest` below uses pointers to slices (of `string`s).

Is that intentional, or would it be better to make it consistently pointers to slices?

I've checked, and the rest of the codebase actually does something similar to what you have here, so maybe this is fine. But I just wanted to raise this to your attention and check what you think.